tests/functional/afr/test_client_io_threads_on_replicated_volumes.py

```diff
@@ -0,0 +1,166 @@
+        self.assertEqual(ret['performance.client-io-threads'], value,
+                         "performance.client-io-threads value {} instead "
+                         "of {}".format(ret['performance.client-io-threads'],
+                                        value))
+
+        # Check if io-threads is loaded or not based on enabled param value
+        ret = occurences_of_pattern_in_file(
+            self.mnode, 'io-threads', "/var/lib/glusterd/vols/{}/trusted-{}."
+            "tcp-fuse.vol".format(self.volname, self.volname))
+        self.assertEqual(ret, instances, "Number of io-threads more than {}"
+                         .format(instances))
+
+    def test_client_io_threads_on_replicate_volumes(self):
+        """
+        Test case 1:
+        1. Create distrubuted volume and start it.
+        2. Check the value of performance.client-io-threads it should be ON.
+        3. io-threads should be loaded in trusted-.tcp-fuse.vol.
+        4. Add brick to convert to replicate volume.
+        5. Check the value of performance.client-io-threads it should be OFF.
+        6. io-threads shouldn't be loaded in trusted-.tcp-fuse.vol.
+        7. Remove brick so thate volume type is back to distributed.
+        8. Check the value of performance.client-io-threads it should be ON.
+        9. performance.client-io-threads should be loaded in
+           trusted-.tcp-fuse.vol.
+
+        Test case 2:
+        1. Create a replicate volume and start it.
+        2. Set performance.client-io-threads to ON.
+        3. Check the value of performance.client-io-threads it should be ON.
+        4. io-threads should be loaded in trusted-.tcp-fuse.vol.
+        5. Add bricks to convert to make the volume 2x3.
+        6. Check the value of performance.client-io-threads it should be ON.
+        7. io-threads should be loaded in trusted-.tcp-fuse.vol.
+        8. Remove brick to make the volume 1x3 again.
+        9. Check the value of performance.client-io-threads it should be ON.
+        10. performance.client-io-threads should be loaded in
+            trusted-.tcp-fuse.vol.
+        """
+        # If volume type is distributed then run test case 1.
+        if self.volume_type == "distributed":
+
+            # Check the value of performance.client-io-threads it should be ON
+            # and io-threads should be loaded in trusted-.tcp-fuse.vol
+            self._check_value_of_performance_client_io_threads()
+
+            # Add brick to convert to replicate volume
+            brick = form_bricks_list_to_add_brick(self.mnode, self.volname,
+                                                  self.servers,
+                                                  self.all_servers_info)
+            self.assertIsNotNone(brick,
+                                 "Failed to form brick list to add brick")
+
+            ret, _, _ = add_brick(self.mnode, self.volname, brick,
+                                  force=True, replica_count=2)
```

Comment:
Can you expand it to replica 3 as replica 2 is the least used and not recommended configuration?

Reply:
By default the testcase is written to check if performance.io-threads is set automatically to ON once we make a pure distrep volume to 1x2. I do understand that we no longer test 1x2 but if I change it 1x2 I feel it'll create a gap where can have issues with 1x2.

Reply:
Alright, keep it as is.

Reply:
@kshithijiyer we are not recommending customers to to use replica 2 volumes. So i feel we should use *3 configuration.

Reply:
The testcase is designed to check if the moment we change a dist volume to replicate volume it should show that the options to off. Now if I go by your comment and change it replicate 1x3 there is a high chance that someday it'll fail for 1x2 but pass in 1x3, this defies the purpose of the testcase.

Reply:
okay fine.